### PR TITLE
Handle the case where complement has a GCS

### PR DESCRIPTION
**Pull request by j9ac9k**
When the compliment dataset has a global coordinate system, care must be taken to project that system into a PCS before registration can take place.  This is largely due to codem being dependent on the resolution for the transformation matrix associated with the dataset (in a GCS the units are degrees, not meters/feet).

### CODE REVIEW

File: src/codem/preprocessing/preprocess.py [hunk 1]
```diff
@@ -134,6 +136,21 @@ def resolution(self, value: float) -> None:
         self._resolution = value
         return None
 
+    def _PCS_from_GCS(
```
Comment (hobu) [inline src/codem/preprocessing/preprocess.py:139]: This isn't a great name for this function. What you're doing here is computing a possible UTM zone in which to do your operations.

File: src/codem/preprocessing/preprocess.py [hunk 2]
```diff
@@ -408,6 +429,30 @@ def _create_dsm(self, resample: bool = True) -> None:
             self.nodata = data.nodata
             self.crs = data.crs
 
+            if not self.fnd and self.crs is not None and not self.crs.is_projected:
```
Comment (hobu) [inline src/codem/preprocessing/preprocess.py:432]: 🪢 🪢 🪢

File: src/codem/preprocessing/preprocess.py [hunk 3]
```diff
@@ -547,39 +617,44 @@ def _calculate_resolution(self) -> None:
...
         """
-        pdal_pipeline = [
-            self.file,
-            {"type": "filters.hexbin", "edge_size": 25, "threshold": 1},
-        ]
-        pipeline = pdal.Pipeline(json.dumps(pdal_pipeline))
+        tag = ["AOI", "Foundation"][int(self.fnd)]
+
+        pipeline = pdal.Reader(self.file)
+        pipeline |= pdal.Filter.hexbin(edge_size=25, threshold=1)
```
Comment (hobu) [inline src/codem/preprocessing/preprocess.py:623]: Someday we should make that `edge_size` configurable. It's probably too coarse for drone collects and the like.

Comment (j9ac9k) [reply src/codem/preprocessing/preprocess.py:623]: Going to create a feature request issue from this comment...

File: src/codem/preprocessing/preprocess.py (hunk 2 above)
Comment (j9ac9k) [reply src/codem/preprocessing/preprocess.py:432]: why tie a knot when you can tie a lot...

Comment (j9ac9k) [reply src/codem/preprocessing/preprocess.py:432]: Should be a bit better; although to accommodate mypy, still needed to keep one `and` in there.

**Pull request merged**